executor/builder.go

```diff
@@ -1075,7 +1075,7 @@ func (b *executorBuilder) buildHashAgg(v *plannercore.PhysicalHashAgg) Executor
 				ordinal = append(ordinal, partialOrdinal+1)
 				partialOrdinal++
 			}
-			finalDesc := aggDesc.Split(ordinal)
+			finalDesc := aggDesc.Split(ordinal, sessionVars.StmtCtx.UseCache)
```

Comment:
Could you explain the root cause a little bit more? Why would `UseCache` impact the behavior of executor?